client/extensions/woocommerce/app/products/product-form.js

```diff
@@ -51,16 +51,38 @@ export default class ProductForm extends Component {
 				icon=""
 				expanded={ true }
 				className="product-variations"
-				header={ ( <FormToggle onChange={ this.handleToggle } checked={ this.state.isVariation }>
+				header={ ( <FormToggle onChange={ this.handleVariationToggle } checked={ 'variable' === product.type }>
 				{variationToggleDescription}
 				</FormToggle>
 				) }
 			>
-				{ this.state.isVariation && (
-					<ProductVariationTypesForm />
+				{ 'variable' === product.type && (
+					<ProductVariationTypesForm
+						product={ product }
+						editProduct={ this.props.editProduct }
+					/>
 				) }
 			</FoldableCard>
 		);
 	}
 
 }
+
+function mapStateToProps( state ) {
+	const { products } = state.extensions.woocommerce;
+	// @TODO Load in proper product when editing an existing product.
+	return {
+		product: products.edits.add,
```

Comment:
I'd like to see the `products.edits.add` part of the reducer go away. I don't see value in creating defaults and holding them in state when they could just be created on demand with a function.